Add a mechanism to rename columns

set-soft opened the pull request:
## Description

Column names displayed in the table head are the names of the KiCad fields or the internal names used by KiBoM.
Sometimes the names aren't pretty. As an example is common to use `manf#` for the part name used by the component manufacturer.
Using long names for KiCad fields is a bad idea.

This patch allows to rename the columns names so you can use pretty names in the headings.

You must define a `COLUMN_RENAME` section in the configuration file.
Then add entries with the name of the original field and the name you want, separated by a tab (ASCII 9).
Here is an example:

```
[COLUMN_RENAME]
manf#	Manufacturer part number
```

Note that you must use a tab as separator. Using a space won't work because *Manufacturer part number* also has spaces.

## Limitations

The separator must be a tab.

## Additional notes

- This patch also makes the `prefs.ignore` list lowercase.
  This is to avoid converting it to lower case all the time.
- `prefs.colRename` is a hash. I saw that various preference options should also be a *dict* instead of a *list*.

WaeCo commented:
what is the hold up here? 
I would really like to see this merged

set-soft commented:
Perhaps @SchrodingersGat doesn't really like my implementation. But you can always use:

1. My [fork](https://github.com/INTI-CMNB/KiBoM), I try to keep it synced with this repo. So you can think about it as the *experimental* branch.
2. The [KiBot](https://github.com/INTI-CMNB/KiBot) internal BoM output. It has more options and flexibility. But it also has a lot of limitations (only Linux, not a plug-in).

SchrodingersGat commented:
@set-soft sorry for the delay on this, it had completely fallen off my radar. 

If you can please fix the conflicts with `preferences.py` I would be happy to merge.

BillyPocketNC commented:
@set-soft @SchrodingersGat ,
What needs to change to get this merged?

set-soft commented:
Hi @BillyPocketNC !

> @set-soft @SchrodingersGat ,
> What needs to change to get this merged?

My fault, I didn't notice @SchrodingersGat asked for conflict resolution.
I rebased the patch, fixed the conflicts and also fixed the flake8 error introduced by #145 
Now the code passes flake8 tests and doesn't have any conflict.

SchrodingersGat commented:
Thanks :)

BillyPocketNC commented:
Thanks this will be useful 👍
